### PR TITLE
fix(ci): Make sure evergreen runs

### PR DESCRIPTION
**Pull request by gribnoysup**
Merging #2173 #2174 caused evergreen to start failing. The gist of the issues causing this was me doing a poor job of cherry-picking commits for #2174 and missing some fixes in the PR, but also a few other things that this PR tries to address:

- `download-akzidenz` script that we have in every compass plugin caused a lot of flakiness, so I made the script a bit more fail-proof
- evergreen ci flow required to `save` and `restore` the whole repo between every step on most machines which worked fine while compass was a single package, now when it's a monorepo, this project state persisting just doesn't scale. As there seems to be no better way to preserve project state between tasks, change the config to to every run in one go. Not optimal, but I couldn't find a better way to do it with evergreen. Maybe if we are able to hoist/dedupe all deps and use GitHub Actions for testing, we can only use evergreen only for compass packaging
- misc fixes: bigger timeouts for some tests, make sure that test scripts in packages are self-contained, eslintignore build artifacts, etc.

Opening it as a draft for now, the pipeline is still red, but I am getting close and having a draft PR makes it easier for me to track progress and applied changes

### CODE REVIEW

**Comment by gribnoysup** [comment]
As discussed with @mcasimir this PR more or less managed to address most of the overall monorepo related issues with evergreen CI config, now it's individual tests cases in packages that are failing. To prevent the diff from growing even bigger, we are scoping evergreen tasks back to only run on compass (see 1d08d9e) so that we can unblock release pipeline and merge this, in the meantime I'll compile a list of all failing tests and open separate tasks for them and will continue fixing them one by one.

File: packages/storage-mixin/test/helpers.js [hunk 1]
```diff
@@ -91,11 +92,47 @@ var Users = Collection.extend({
...
+    const tempKeychainName = `temp${Date.now().toString(32)}.keychain`;
+    const origDefaultKeychain = JSON.parse(getDefaultKeychain());
+
+    return {
+      name: tempKeychainName,
+      before() {
+        execSync(`security create-keychain -p "" "${tempKeychainName}"`);
+        execSync(`security default-keychain -s "${tempKeychainName}"`);
+        execSync(`security unlock-keychain -p "" "${tempKeychainName}"`);
+        execSync(`security set-keychain-settings "${tempKeychainName}"`);
```
Comment (gribnoysup) [inline packages/storage-mixin/test/helpers.js:112]:
```suggestion
```

**Comment by gribnoysup** [comment]
Green evergreen spawned with latest changes from this PR: https://spruce.mongodb.com/version/6091313a61837d655b403d6a/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC

**Review by Anemy — APPROVED** [review]
looked through this together, lgtm
nice work diving into some of these evergreen issues.
Getting our mocha and some other dev deps versions aligned might solve a failing test or two. Looking into that now.
These are our mocha versions lol
```
{
    '^2.5.3',
    '^3.0.2',
    '^3.1.2',
    '^3.2.0',
    '^4.1.0',
    '^5.0.0',
    '^5.2.0',
    '^6.0.2',
    '^6.1.2',
    '^6.2.2',
    '^7.0.0',
    '^7.0.1',
    '^7.1.0',
    '^7.1.1',
    '^8.0.1',
    '^8.1.3',
    '^8.2.1',
    '^8.3.2'
}
```

File: .evergreen.yml [hunk 2]
```diff
@@ -387,36 +396,68 @@ functions:
...
+
       - func: install
-      # - func: test
-      #   vars:
-      #     test_suite: functional
+
+      - func: verify
+
+      - func: test
+        # TODO: rhel pretest fails and needs some additional investigation to make it work
```
Comment (gribnoysup) [inline .evergreen.yml:411]: Opened a ticket for that https://jira.mongodb.org/browse/COMPASS-4789

File: .github/workflows/pr.yaml [hunk 3]
```diff
@@ -26,14 +26,17 @@ jobs:
...
 
-      - name: Install Deps Ubuntu
-        run: sudo apt-get -y install libkrb5-dev libsecret-1-dev net-tools libstdc++6 gnome-keyring
+      # - name: Install Deps Ubuntu
+      #   run: sudo apt-get -y install libkrb5-dev libsecret-1-dev net-tools libstdc++6 gnome-keyring
 
-      - name: Install Dependencies
-        run: npm ci
+      - name: Not Implemented
+        run: echo "GitHub Actions workflows are disabled for now. To be fixed in https://jira.mongodb.org/browse/COMPASS-4788"
```
Comment (gribnoysup) [inline .github/workflows/pr.yaml:33]: See also https://github.com/mongodb-js/compass/pull/2174#issuecomment-829264752. Opened a ticket for that https://jira.mongodb.org/browse/COMPASS-4788

**Pull request merged**